internal/backend/remote-state/http/helper.go

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) The OpenTofu Authors
+// SPDX-License-Identifier: MPL-2.0
+// Copyright (c) 2023 HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package http
+
+import (
+	"encoding/json"
+	"io"
+	"log"
+	"net/http"
+)
+
+// Helper function to parse response body bytes into a string for logging
+func parseResponseBody(resp *http.Response) string {
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		log.Printf("[ERROR] Failed to read HTTP response body for Logging: %v", err)
+		return ""
+	}
+	return string(body)
+}
+
+// Helper function to parse http headers into a json string for logging
+func parseHeaders(resp *http.Response) string {
+	// Blacklist of Header keys that need to be masked
+	var blacklist = map[string]bool{
+		"Authorization": true,
+		"Cookie":        true,
+		"Set-Cookie":    true,
+	}
+
+	headers := make(map[string]string, len(resp.Header))
+	for key := range resp.Header {
```

Comment:
HTTP headers are case-insensitive so we must normalize the values before comparison. Also, headers in the blacklist map should be already normalized (e.g. in lowercase).